libs/extensions/std/exec/src/gtfs-rt-interpreter-executor.ts

```diff
@@ -0,0 +1,304 @@
+import { IOType } from '@jvalue/jayvee-language-server';
+import * as E from 'fp-ts/lib/Either';
+import { Either } from 'fp-ts/lib/Either';
+import * as GtfsRealtimeBindings from 'gtfs-realtime-bindings';
+
+@implementsStatic<BlockExecutorClass>()
+export class GtfsRTInterpreterExecutor
+  implements BlockExecutor<IOType.FILE, IOType.SHEET>
+{
+  public static readonly type = 'GtfsRTInterpreter';
+  public readonly inputType = IOType.FILE;
+  public readonly outputType = IOType.SHEET;
+
+  async execute(
+    inputFile: BinaryFile,
+    context: ExecutionContext,
+  ): Promise<R.Result<Sheet>> {
+    // Accessing attribute values by their name:
+    const entity = context.getTextPropertyValue('entity');
+
+    // https://github.com/MobilityData/gtfs-realtime-bindings/tree/master/nodejs
+    const feedMessage =
+      GtfsRealtimeBindings.transit_realtime.FeedMessage.decode(
+        new Uint8Array(inputFile.content),
+      );
+
+    // Parse all possible feedentity to Sheet
+    const sheet = await this.parseFeedMessage(entity, feedMessage, context);
+    if (E.isLeft(sheet)) {
+      return Promise.resolve(
+        R.err({
+          message: sheet.left.message,
+          diagnostic: { node: context.getCurrentNode(), property: 'name' },
+        }),
+      );
+    }
+    return R.ok(sheet.right);
+  }
+
+  private async parseFeedMessage(
+    entity_type: string,
+    feedMessage: GtfsRealtimeBindings.transit_realtime.FeedMessage,
+    context: ExecutionContext,
+  ): Promise<Either<Error, Sheet>> {
+    return new Promise((resolve) => {
+      switch (entity_type) {
+        case 'trip_update':
+          // Extract the trip updates from thee feed message
+          resolve(this.parseTripUpdates(feedMessage, context));
+          break;
+        case 'alert':
+          // Extract the alerts from thee feed message
+          resolve(this.parseAlerts(feedMessage, context));
+          break;
+        case 'vehicle':
+          // Extract vehicle positions from thee feed message
+          resolve(this.parseVehiclePositions(feedMessage, context));
+          break;
+
+        // Entity invalid
+        default:
+          resolve(
+            E.left(
+              new Error(
+                `Entity should be either "trip_update", "alert" or "vehicle". Please check the argument "entity" of the GtfsRTInterpreterblock.`,
+              ),
+            ),
+          );
+          break;
+      }
+    });
+  }
+
+  private parseTripUpdates(
+    feedMessage: GtfsRealtimeBindings.transit_realtime.FeedMessage,
+    context: ExecutionContext,
+  ): Promise<Either<Error, Sheet>> {
+    return new Promise((resolve) => {
+      context.logger.logDebug(`Parsing raw gtfs-rt feed data as TripUpdate"`);
+      const rows: string[][] = [];
+
+      // Add Header
+      rows.push([...tripUpdateHeader]);
+
+      for (const entity of feedMessage.entity) {
+        if (entity.tripUpdate) {
+          if (entity.tripUpdate.stopTimeUpdate) {
```

Comment:
You can avoid deep nesting by negating the if conditions, e.g.
```
if (!entity.tripUpdate) {
  // log that tripUpdate was not present
  continue;
}
if (!entity.tripUpdate.stopTimeUpdate) {
  // log that stopTimeUpdate was not present
  continue;
}
// ...
```

The same technique can also be applied to your other functions.